File: ios/Runner/MethodChannelDemo.swift
//
//  MethodChannelDemo.swift
//  Runner
//
//  Created by a on 2020/12/28.
//

import Foundation

import Flutter
import UIKit

public class MethodChannelDemo {
    init(messenger: FlutterBinaryMessenger) {
        let channel = FlutterMethodChannel(name: "com.mufeng.flutterNativeChannel.MethodChannel", binaryMessenger: messenger)
        channel.setMethodCallHandler{ (call: FlutterMethodCall, result:@escaping FlutterResult)in
            if(call.method == "add"){
                if let dict = call.arguments as? Dictionary<String, Any>{
                    let params1: Int = dict["params1"] as? Int ?? 0
                    let params2: Int = dict["params2"] as? Int ?? 0
                    let params3: Int = dict["params3"] as? Int ?? 0
                    result(params1 + params2 + params3)
                }
            }
        }
    }
}

File: android/app/src/main/kotlin/com/mufeng/flutter_native_channel/MethodChannelDemo.kt
package com.mufeng.flutter_native_channel

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

class MethodChannelDemo (messenger: BinaryMessenger): MethodChannel.MethodCallHandler{

    private var channel: MethodChannel = MethodChannel(messenger, "com.mufeng.flutterNativeChannel.MethodChannel")

    init {
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when(call.method){
            "add" -> {
                val params1 = call.argument<Int>("params1")
                val params2 = call.argument<Int>("params2")
                val params3 = call.argument<Int>("params3")
                val params = params1!! + params2!! + params3!!
                result.success(params)
            }
        }
    }
}